shared/security/src/main/java/io/pravega/shared/security/auth/PermissionsHelper.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.security.auth;
+
+import com.google.common.base.Strings;
+import io.pravega.auth.AuthHandler;
+import lombok.NonNull;
+
+/**
+ * Helper methods for permissions.
+ */
+public class PermissionsHelper {
+
+    /**
+     * Translates the specified {@code accessOperation} to an {@link AuthHandler.Permissions} object.
+     *
+     * @param accessOperation accessOperation to translate
+     * @return a {@link AuthHandler.Permissions} object that represents the specified {@code accessOperation}
+     */
+    public static AuthHandler.Permissions toAuthHandlerPermissions(@NonNull AccessOperation accessOperation) {
+        if (accessOperation.equals(AccessOperation.READ)) {
+            return AuthHandler.Permissions.READ;
+        } else if (accessOperation.equals(AccessOperation.WRITE) || accessOperation.equals(AccessOperation.READ_WRITE)) {
+            return AuthHandler.Permissions.READ_UPDATE;
+        } else {
+            throw new IllegalArgumentException("Cannot translate access operation " + accessOperation.name());
+        }
+    }
+
+    /**
+     * Parse the specified {@code accessOperationStr} string and translate it to an {@link AuthHandler.Permissions} object.
+     *
+     * @param accessOperationStr a string value of an {@link AccessOperation} object
+     * @param defaultPermissions the default {@link AuthHandler.Permissions} object to return in case the
+     *                           {@code accessOperationStr} can't be parsed.
+     * @return the parsed or default {@link AuthHandler.Permissions} object,
+     */
+    public static AuthHandler.Permissions parse(String accessOperationStr, AuthHandler.Permissions defaultPermissions) {
+        if (Strings.isNullOrEmpty(accessOperationStr)) {
+            return defaultPermissions;
+        }
+        try {
+            return toAuthHandlerPermissions(AccessOperation.valueOf(accessOperationStr));
+        } catch (IllegalArgumentException e) {
+            return defaultPermissions;
```

Comment:
I'm wondering whether returning the default permissions is the right thing to do. If the string is illegal and we return the default, then aren't we implicitly assuming that the default is what the caller wants? It might be a better approach to error it out.

Reply:
This method is quite like the Java [System.getProperty​(String key, String def)](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/System.html#getProperty(java.lang.String,java.lang.String)). It takes a default value as input and returns that if it is unable to parse. The usage of this method is in `StreamAuthParams`, which sets the default as `READ`, based on the purpose it in turn is used in `ControllerServiceImpl`. 

Does that sound good to you?

Reply:
Say we are trying to get a permission to read_update, and for some reason the string is messed up. Say that the default permission is read, in which case this is returning a read permission. Assuming a scenario like this is possible,  transparently returning the default sounds problematic because that's not the intention.

Reply:
Ok. I've modified the code to throw the exception as-is when a value is present but cannot be parsed.